src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,220 @@
+            numberOfElements++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Node<T> result = giveNodeByIndex(index);
+        return result.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> oldNode = giveNodeByIndex(index);
+        Node<T> newNode = new Node<>(null, value, null);
+        changeConnectionFromToNode(oldNode, newNode);
+        return oldNode.item;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> node = giveNodeByIndex(index);
+        disconnectNode(node);
+        numberOfElements--;
+        return node.item;
     }
 
     @Override
     public boolean remove(T object) {
+        int count = 0;
+        Node<T> node;
+        while (count < numberOfElements) {
+            node = giveNodeByIndex(count);
+            if (node.item == null
+                    ? node.item == object : node.item.equals(object)) {
+                remove(count);
+                return true;
+            }
+            count++;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return numberOfElements;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return numberOfElements == 0;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("[ ");
+        Node<T> node;
+        int counter = 0;
+        while (counter < size()) {
+            node = giveNodeByIndex(counter);
+            stringBuilder.append(node);
+            if (counter + 1 < size()) {
+                stringBuilder.append(", ");
+            }
+            counter++;
+        }
+        stringBuilder.append(" ]");
+        return stringBuilder.toString();
+    }
+
+    private Node<T> giveNodeByIndex(int index) {
+        if (index < 0 || index > numberOfElements) {
+            throw new IndexOutOfBoundsException("Wrong index");
+        }
+        Node<T> result;
+        if (index > numberOfElements / 2) {
+            result = giveNodeByIndexFromTail(index);
+        } else {
+            result = giveNodeByIndexFromHead(index);
+        }
+        return result;
+    }
+
+    private Node<T> giveNodeByIndexFromTail(int index) {
+        Node<T> node = tail;
+        int counter = numberOfElements;
+
+        while (node != null) {
+            counter--;
+            if (counter == index) {
+                return node;
+            }
+            node = node.previous;
+
+        }
+        throw new IndexOutOfBoundsException("Wrong index");
+    }
+
+    private Node<T> giveNodeByIndexFromHead(int index) {
+        Node<T> node = head;
+        int counter = 0;
+
```

Comment:
Empty line

Reply:
This blank line uses for create logical grouping, to split variable initialization, and working code.
According to https://mate-academy.github.io/style-guides/java/java.html#s4.6.1-vertical-whitespace